scripts/globals/pasteventwatcher.lua

```diff
@@ -0,0 +1,220 @@
+-----------------------------------
+-- Past Event Watcher NPCs
+-----------------------------------
+
+-----------------------------------
+
+require("scripts/globals/quests")
+require("scripts/globals/missions")
+require("scripts/globals/zone")
+
+pasteventwatcher = {}
+
+local gobCS = -- add the goblin cs to this table
+{
+    [102] = 126, -- La Theine Plateau
+    [106] = 247, -- North Gustaberg
+    [107] = 905, -- South Gustaberg
+    [119] = 35,  -- Meriphataud Mountains
+    [122] = 6,   -- Ro'Maeve
+    [162] = 10,  -- Castle Zvahl Keep
+    [167] = 100, -- Bostaunieux Oubliette
+    [198] = 67,  -- Maze of Shakhrami
+    [200] = 61,  -- Garlaige Citadel
+}
+
+function pasteventwatcher.rewatch(player, trigger)
+    local zone = player:getZoneID()
+    local ID = zones[zone]
+    local options = 1
+    local cs = -- add checks to this table
```

Comment:
Can we move this outside of the function? The table doesn't need to be constructed every time someone watches a cutscene, just reference it.

Reply:
I based this on the checkReqs function in bcnm.lua. It can be moved out but I'm not sure how to make it look cleaner than doing something like
```lua
    [dsp.zone.MERIPHATAUD_MOUNTAINS] =
    {
        reqs = function(player, i)
            return switch (i): caseof
            {
                [1] = function() return (player:hasCompletedQuest(SANDORIA, dsp.quest.id.sandoria.THE_HOLY_CREST)) end,
            }
        end,
    },
```
to use the player.

Reply:
cutscene moved outside function.